components/woocommerce/sources/common/base.mjs

```diff
@@ -0,0 +1,80 @@
+import woocommerce from "../../woocommerce.app.mjs";
+import crypto from "crypto";
+
+export default {
+  props: {
+    woocommerce,
+    db: "$.service.db",
+    http: "$.interface.http",
+    topics: {
+      propDefinition: [
+        woocommerce,
+        "topics",
+      ],
+    },
+  },
+  hooks: {
+    async activate() {
+      const hookIds = [];
+      for (const topicType of this.topics) {
+        const topic = this.getTopic(topicType);
+        const data = {
+          topic,
+          delivery_url: this.http.endpoint,
+          secret: this.woocommerce.$auth.secret,
+        };
+        const { id } = await this.woocommerce.createWebhook(data);
+        hookIds.push(id);
+      }
+      this._setHookIds(hookIds);
+    },
+    async deactivate() {
+      const hookIds = this._getHookIds();
+      for (const id of hookIds) {
+        await this.woocommerce.deleteWebhook(id);
+      }
+    },
```

Comment:
**Suggestion/Question?**
DB reset after deleting hooks.

Since the source will be removed is this necessary or is it automatic?

```suggestion
      this._setHookIds(null);
    },
```